doc: Entity PATCH API to update entity definition (#3588)
thin-edge/thin-edge.io: docs/src/operate/registration/register.md

> In addition to the MQTT APIs, %%te%% now supports the following REST APIs as well to manage entities with it:

Author: Resolved in b364661. Overwrote the previous commit to avoid another squash and push.

* Creation
* Retrieval

> * Updation

Reviewer: Google says it's Indian English :D I think we can use update as noun.

```suggestion
* Update
```

> * Updation

Author: > Google says it's Indian English :D

I think we introduced that word to better rhyme with creation and deletion 😉 

Anyway, rewrote that sentence in b364661

    }
]
```
## Update entity

An existing entity can be updated using the PATCH API.
But the updates are limited to the `@parent` and `@health` endpoints only.
Other properties like `@type` and `@id` can not be updated after the registration.

> The `@parent` must be another device, as services can not ave children.

Reviewer 2: ```suggestion
The `@parent` must be another device, as services cannot have children.
```

Author: Resolved in 3f85650

    }
]
```
## Update entity

An existing entity can be updated using the PATCH API.
But the updates are limited to the `@parent` and `@health` endpoints only.
Other properties like `@type` and `@id` can not be updated after the registration.

**Endpoint**

```
PATCH /v1/entities/{topic-id}
```

**Payload**

```json
{
    "@parent": "{new-parent-topic-id}",

>     "@health": "{some-service-topic-id}"

Author: Resolved in b364661. Overwrote the previous commit to avoid another squash and push.
